src/ios/CDVNotification.m

```diff
@@ -39,100 +39,49 @@ @implementation CDVNotification
-                }
+    for (int n = 0; n < count; n++) {
+        [alertController addAction:[UIAlertAction actionWithTitle:[buttons objectAtIndex:n]
+                                                            style:UIAlertActionStyleDefault
+                                                          handler:^(UIAlertAction * action)
+        {
+            CDVPluginResult* result;
 
-                [weakNotif.commandDelegate sendPluginResult:result callbackId:callbackId];
-            }]];
-        }
-        
-        if ([dialogType isEqualToString:DIALOG_TYPE_PROMPT]) {
-            
-            [alertController addTextFieldWithConfigurationHandler:^(UITextField *textField) {
-                textField.text = defaultText;
-            }];
-        }
-        
-        if(!alertList)
-            alertList = [[NSMutableArray alloc] init];
-        [alertList addObject:alertController];
-        
-        if ([alertList count]==1) {
-            [self presentAlertcontroller];
-        }
-        
-    }
-    else
-    {
-#endif
+            if ([dialogType isEqualToString:DIALOG_TYPE_PROMPT])
+            {
+                NSString* value0 = [[alertController.textFields objectAtIndex:0] text];
+                NSDictionary* info = @{
+                    @"buttonIndex":@(n + 1),
+                    @"input1":(value0 ? value0 : [NSNull null])
+                };
+                result = [CDVPluginResult resultWithStatus:CDVCommandStatus_OK messageAsDictionary:info];
+            }
+            else
+            {
+                result = [CDVPluginResult resultWithStatus:CDVCommandStatus_OK messageAsInt:(int)(n  + 1)];
+            }
 
-        CDVAlertView* alertView = [[CDVAlertView alloc]
-                                   initWithTitle:title
-                                   message:message
-                                   delegate:self
-                                   cancelButtonTitle:nil
-                                   otherButtonTitles:nil];
-        
-        alertView.callbackId = callbackId;
-        
-        
-        
-        for (int n = 0; n < count; n++) {
-            [alertView addButtonWithTitle:[buttons objectAtIndex:n]];
-        }
+            [weakNotif.commandDelegate sendPluginResult:result callbackId:callbackId];
+        }]];
+    }
+    if ([dialogType isEqualToString:DIALOG_TYPE_PROMPT]) {
         
-        if ([dialogType isEqualToString:DIALOG_TYPE_PROMPT]) {
-            alertView.alertViewStyle = UIAlertViewStylePlainTextInput;
-            UITextField* textField = [alertView textFieldAtIndex:0];
+        [alertController addTextFieldWithConfigurationHandler:^(UITextField *textField) {
             textField.text = defaultText;
-        }
-        
-        [alertView show];
-#ifdef __IPHONE_8_0
+        }];
+    }
+    if(!alertList)
+        alertList = [[NSMutableArray alloc] init];
+    [alertList addObject:alertController];
```

Comment:
```suggestion
        [alertList addObject:alertController];
```

Indentation seems of on this line.

Reply:
no, indentation is ok, but the if should probably have curly braces

Reply:
added them, doesn't look off now, right?

Reply:
👍